### PR TITLE
:bug: update col_types for breakpoints; cutoff 100

### PR DESCRIPTION
**Pull request by kgaonkar6**
Col_type detection by readr was in-correct (date instead of charatcer) issue in annoFusePerSample.R  causes breakpoints to have NA in output.

This will need re-runs for annoFuse in kfdrc workflows

### CODE REVIEW

**Review by dmiller15 — CHANGES_REQUESTED** [review]
The changes are fine but, because there were no changes to the Dockerfile, it was not built and pushed. You can either push the new Docker image yourself or modify the Dockerfile here for it to be done automatically.

**Comment by kgaonkar6** [comment]
@dmiller15 I updated the dockerfile with a comment, let me know if that works

**Comment by dmiller15** [comment]
Looks like there's a bug in the config. It did build successfully though so I'll approve.

**Review by dmiller15 — APPROVED** [review]

**Comment by kgaonkar6** [comment]
> Looks like there's a bug in the config. It did build successfully though so I'll approve.

https://hub.docker.com/repository/docker/gaonkark/annofuse it seems it didn't get built  right?  from the check it seems because the file was named dockerfile, I've updated it to Dockerfile.

File: annoFuse/0.1.8/annoFusePerSample.R [hunk 1]
```diff
@@ -89,7 +89,7 @@ if(!is_empty(STARFusioninputfile$FusionName) & is_empty(Arribainputfile$"gene1--
 
 
 # General fusion QC for read support and red flags
-fusionQCFiltered<-fusion_filtering_QC(standardFusioncalls=standardFusioncalls,readingFrameFilter="in-frame|frameshift|other",artifactFilter="GTEx_Recurrent|DGD_PARALOGS|Normal|BodyMap|ConjoinG",junctionReadCountFilter=1,spanningFragCountFilter=10,readthroughFilter=FALSE)
+fusionQCFiltered<-fusion_filtering_QC(standardFusioncalls=standardFusioncalls,readingFrameFilter="in-frame|frameshift|other",artifactFilter="GTEx_Recurrent|DGD_PARALOGS|Normal|BodyMap|ConjoinG",junctionReadCountFilter=1,spanningFragCountFilter=100,readthroughFilter=FALSE)
```
Comment (migbro) [inline annoFuse/0.1.8/annoFusePerSample.R:92]: hmmm, why the change from 10 to 100? I've been out of the loop a while, seems some new stuff has popped up. Also, if a default filter param has changed, then perhaps it should be considered a minor version update? like 0.1.9?

**Comment by migbro** [comment]
> > Looks like there's a bug in the config. It did build successfully though so I'll approve.
> 
> https://hub.docker.com/repository/docker/gaonkark/annofuse to me it seems it didn't get built right? from the check is it because the file was named dockerfile, I've updated it to Dockerfile.
> 
> Maybe not

@dmiller15 , was it the dockerfile vs Dockerfile and/or the script/circleCI doesn't like to build with paths that have uppercase letters?

**Comment by dmiller15** [comment]
@migbro From the test:
```
invalid argument "docker.io/kfdrc/annoFuse:0.1.8" for "-t, --tag" flag: invalid reference format: repository name must be lowercase
See 'docker build --help'.
```
It failed because the repository name needs to be lower case. It failed before it had a chance to build. The build may have failed. It depends on if the docker build statement would have picked up the lower case dockerfile. Of course even if it succeeded it would have created issues down the line so recapitalizing it is the correct approach.

The fix for this went in in a #75

File: annoFuse/0.1.8/annoFusePerSample.R (hunk 1 above)
Comment (kgaonkar6) [reply annoFuse/0.1.8/annoFusePerSample.R:92]: With the cutoff of 10 we were able to get ~96% sensitivity for PPTC (100 read length ) data.  However, when we tested TCGA data which has (50-75 read length) the cutoff of 10 wasn't sufficient (~70% sensitivity) but at 100 we get about ~96% sensitivity (percentage overlap to what is previously published for these samples)
Screen Shot 2020-09-11 at 2 30 06 PM

So we believe we should change the cutoff to 100 to capture all read length datasets , tagging @jharenza.

I can update to the latest tag of the package as well which currently is v0.2, if that seems necessary here.

Comment (migbro) [reply annoFuse/0.1.8/annoFusePerSample.R:92]: Oooh, interesting! What about the specificity? Are you saying performance stayed good on the original dataset while improving TCGA results? Or should the cutoff move with input read length?

Comment (kgaonkar6) [reply annoFuse/0.1.8/annoFusePerSample.R:92]: So yeah the overlap of high confidence fusion in PPTC dataset remained the same at spanningFragCountFilter == 100 but changing the cutoff to 100 did significantly affect the overlap with TCGA fusion high confidence fusion list.

Technically yes it seems the cutoff is affected by read length but we don't have clear cutoff for each type of read length dataset ( 50,100,150 etc) nor do we see any published paper on that.  So going by TCGA data overlap as reference the 100 cutoff might be best.

About specificity we don't have any true negatives (I think that would only come from like lab experiments to add true negative to the samples not sure ) to calculate that.

**Comment by jharenza** [comment]
@migbro TCGA was requested by a reviewer, and glad it was. Moving this to 100 for the longer reads is probably not necessary, and it only added 65 fusions for PBTA. We are going to have to opt for being a little less stringent here as a default, since TCGA and PPTC are the only datasets we have with high-confidence fusion calls and it is hard to know whether 10 or 100 really is best for 150 bp seq. Hopefully, Kristen Dang (SAGE) and I get our R03 grant, which aims to benchmark fusions and would help with this. We added this as a param to enable authors to choose a cutoff as well.

**Comment by kgaonkar6** [comment]
closing this since we are updating to v0.90.0
